packages/electron-updater/src/windowsExecutableCodeSignatureVerifier.ts

```diff
@@ -54,8 +54,21 @@ export function verifySignature(publisherNames: Array<string>, unescapedTempUpda
 
           const data = parseOut(Buffer.from(stdout, "base64").toString("utf-8"))
           if (data.Status === 0) {
-            const name = parseDn(data.SignerCertificate.Subject).get("CN")!
-            if (publisherNames.includes(name)) {
+            const subject = parseDn(data.SignerCertificate.Subject)
+            let match = false
+            publisherNames.forEach(name => {
+              const dn = parseDn(name)
+              if (dn.size) {
+                // if we have a full DN, compare all values
+                const allKeys = Array.from(dn.keys())
+                match = allKeys.every(key => {
```

Comment:
I might be overlooking something, but I'm not sure if this will work when multiple `publisherNames` are provided since we don't return early after `match = true`. If we have two fully defined publisher DNs (`p1` and `p2`), and `p1` passes/matches but then `p2` fails, `match` would be false after this `forEach`
I think we need to do a traditional for-loop so that we can `break` out of it once `match = true`
What are your thoughts?

Reply:
Nope, you're right, that will definitely cause an issue. I think it definitely makes sense to just break the loop here once we're done to make sure the match is set correctly. Nice catch!